airbyte-webapp/src/views/Connector/ServiceForm/ServiceForm.tsx

```diff
@@ -83,9 +84,40 @@ const PatchInitialValuesWithWidgetConfig: React.FC<{ schema: JSONSchema7 }> = ({
   return null;
 };
 
+/**
+ * A component that will observe whenever the serviceType (selected connector)
+ * changes and set the name of the connector to match the connector definition name.
+ */
+const SetDefaultName: React.FC<{
+  availableConnectors: ConnectorDefinition[];
+}> = ({ availableConnectors }) => {
```

Comment:
Instead of passing `availableConnectors` and searching - you could use `useServiceForm` hook that has `selectedService` prop.

So this method will shorten to something like

```
const { selectedService } = useServiceForm();
const [,,helpers] = useField("name");

useEffect(() => {
  const name = selectedService?.name ?? "";
  helpers. setValue(name);
}, [selectedService, helpers]);
```

Reply:
Thanks, I will use `useServiceForm` here.

I tried using `useField` initially, but the problem is that the helpers it's returning isn't a stable reference, thus causing this `useEffect` to trigger in an endless loop, thus using the `setFieldValue` here instead (which is a stable reference).